### ISSUE TITLE
Add note to estimated matching in UI that shows update time (~1 hour)

### ISSUE THREAD

**Issue by koday1** [Giveth/giveth-dapps-v2#4274]
Updating estimated matching after every donation in a QF round requires a lot of resources on the backend. To reduce this load we are shifting to updating the matching every 1 hour instead of after every donation. @RamRamez 

Per our discussion on the QF call today, we need to add a note in the UI next to the matching estimates that says "Matching estimates updated every hour" OR add the timestamps of the last update, and prediction for the next update.

For reference, here is how Gitcoin handles it, with specific timestamps:

<img width="940" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNVpjQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--e4bb5453bb8313845286ac564e3c3b86758b32c2/Gitcoinestimatedmatching.png" alt="Gitcoinestimatedmatching.png" />

@mosaeedi Are you able to help with a design for this note next to estimated matching?

This is related to matching improvements to better represent COCM: https://github.com/Giveth/giveth-dapps-v2/issues/4251

**Comment by laurenluz** [comment]
I love the way the timestamps are being used here in gitcoin and think it would be great to do that! We should probably put a little note about the timestamps in all the places the user sees estimated matching (or at least most of them, including):
- project card
- project single page, matching estimated card
- project single page, donations tab, qf round tab, estimated matching section
- donation page
- donation success page? (i don't remember if we put estimated matching here... I think not)

Also @mosaeedi - keep in mind that we should only show these estimates for current, active QF rounds... not ones that have been completed and have the actual matching shown in qf round tab.

**Comment by mosaeedi** [comment]
**1. Project card**
For Project card, I suggest we put the timestamp inside the tooltip.

<img width="249" alt="Screenshot 2024-07-02 at 8 39 41 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/6c95eea2-9005-4896-8fa9-35f70881347a">


**2. Project single page, matching estimated card**

<img width="875" alt="Screenshot 2024-07-02 at 8 37 59 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/4ff28125-a2c8-49d7-a79f-9eabd86a4dd6">



**3. project single page, donations tab, qf round tab, estimated matching section**

<img width="611" alt="Screenshot 2024-07-02 at 8 42 09 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/9c0f4105-2fec-4ea2-8886-dca76557d1cb">



**4. donation page**
<img width="505" alt="Screenshot 2024-07-03 at 5 21 32 PM" src="https://github.com/Giveth/giveth-dapps-v2/assets/13708322/f190da53-69ca-4342-b8ef-fa5df6181fc8">

5. donation success page
We just tell user their donations are eligible or not! I don't think we need to add the timestamp there!

**Comment by RamRamez** [comment]
For best optimization, we implemented it in way that the update time is dependent on the last donation on the QF round, so if there's no new donation (like in midnight), we won't refresh the estimated matching, but when a new donation is made, refresh estimated matching is called, and there's a debounce time of one hour, so if there are other donations in the meantime, refresh estimated matching won't be called. After one hour, refresh estimated matching will be called on the next donation. So, if there's no new donation, estimated matching is already updated and we've saved our DB processes.

TL;DR
@laurenluz @mosaeedi Is it possible to change the copy to something like "Estimated matching will updated every 60 min"?
because we don't know when exactly the estimated matching will be updated. The only thing we know is that the estimated matching is already updated or it will be updated in less than 60 min.

BTW, @laurenluz I believe we can safely set the estimated matching refresh debounce time to 10 min.

**Comment by laurenluz** [comment]
I think we can remove the "next update" copy... and just put the "last updated" copy. and it solves the problem and simplified the UI.

@mosaeedi 

1. On the project card, you really think we can't put it somewhere? I had this weird and bad experience when I was making donation in the giv-earth round at the beginning... where I made my donation, and then went to the /qf/all page... and then got upset because my donation was there but hte matching said 0 and I thought there was a bug and an issue. I wouldn't have thought to look into the tooltip... can we just make it really small and say "last updated: x"

2. can we make the update copy smaller here so it doesn't detract from the matching numbers? also, let's remove "next update" and show only "last"
![image](https://github.com/Giveth/giveth-dapps-v2/assets/75490971/a4aa9d13-af9c-4603-91c5-7443159a5826)

3. remove "next update"
![image](https://github.com/Giveth/giveth-dapps-v2/assets/75490971/ca3d590a-d075-475c-8e1d-d427179a7a28)

4. remove "next update"
![image](https://github.com/Giveth/giveth-dapps-v2/assets/75490971/7b5b9efc-14db-43a1-85f4-891ced6779b3)

**Comment by laurenluz** [comment]
I just chatted w/ @RamRamez about this... so if we can update the estimated matching every 10 min, we can reduce the extra dev work by just adding a hard-coded note saying... Next update: <10 min - wherever it is applicable.

@mosaeedi what do you think?

Also, is there some way we can put that on the project card? this round when I was donating and things were not updated right away, I thought there was a bug because all the matching said 0 even on projects I donated to... I wouldn't have thought to check the tool tip.

And for 2. can we decrease the font size on the update text? it's stealing the show a bit from the numbers.

**Comment by koday1** [comment]
@laurenluz I like that idea, I see no issues with saying "Next update <10 min" and letting the user check back in to see the update.

I also agree w/ adding this directly on the project card instead of the tooltip. I ran into the same issues with my first donations in the GIV-Earth round.

cc @mosaeedi

**Comment by RamRamez** [comment]
We need this feature for the next QF round.
cc @laurenluz @mosaeedi

**Comment by koday1** [comment]
Hey @RamRamez @mosaeedi just reviving this issue. 

If we will use 10 min as the time for updating estimated matching, can we move forward with creating designs for the notes saying 'Next Update <10 min'? As discussed above, it would be in these 4 places:

1. Project card
2. Project page, matching estimate card
3. Project page, donations tab, qf round tab, estimated matching section (while a round is live)
4. Donation page

**Comment by RamRamez** [comment]
@koday1 it's 2min on production now, maybe we can skip this design?

**Comment by koday1** [comment]
@RamRamez Got it - thanks for letting me know. @laurenluz I'm curious what your thoughts are on this - should we still add a note for the user if we're going with a 2 min update time?

**Comment by laurenluz** [comment]
let's skip it for now. this will all probably change anyway with the updated estiamted matching.

**Comment by divine-comedian** [comment]
@laurenluz is this issue still relevant?

**Comment by laurenluz** [comment]
yes but only in conjuction with the COCM estimated matching update that carlos worked on. we're blocked by a response from the gitcoin team for now. It could be iceboxed, or just on hold

**Comment by laurenluz** [comment]
@mosaeedi can you help me on this issue one more time to update these designs to work with the new changes we made to the donation page? We're working on add this back in now as per https://github.com/Giveth/giveth-dapps-v2/issues/4877

![image](https://github.com/user-attachments/assets/86613e52-8e43-4850-98be-2cc724a3f753)

**Comment by mosaeedi** [comment]
The only difference here is showing the estimated matching in USD?  based on your comment I don't see anything changed on the donation page!

**Comment by laurenluz** [comment]
@mosaeedi no basically on the "donate" page - you had added the timestamp to the old designs, i.e.:

![image](https://github.com/user-attachments/assets/dd75b9e7-e93e-4100-ad42-f1b2dfa8911c)

This design was done before we implemented the little tags "donation will be matched" / "$5 makes you eligible for GIVbacks", i.e.

![image](https://github.com/user-attachments/assets/694bfd34-32d9-4c4c-95ba-d886dbc08d72)

So we need to find the right place for the timestamp with this updated QF design - it's a small thing.

**Comment by mosaeedi** [comment]
Here is the design for showing the estimated matching.
[Link to Figma](https://www.figma.com/design/qXxAEsN3jePyRqBcHR9Ihq/Donation?node-id=6897-5374&t=k9h6BakStnTDjtjx-4)

Here is how it should look:
<img width="784" alt="Screenshot 2025-01-09 at 7 30 22 PM" src="https://github.com/user-attachments/assets/fdb8597c-7195-40ad-91bc-510eb7a06e00" />

**Comment by divine-comedian** [comment]
@yegorgolovnia @laurenluz 

now that the designs are here can you provide a clear description and acceptance criteria for this issue. This will make it easier for the devs to estimate and begin work on this issue

**Comment by divine-comedian** [comment]
Need to get information and save it to a query in order for front-end to get it

**Comment by laurenluz** [comment]
We have implemented a new version of estimated matching that uses COCM via #4877 

The python script that is returning the value for estimated matching for each project runs periodically. In order to reduce confusion for users who make a donation and don't see changes to the estimated matching right away, we should add a "last updated" time to every place we show the estimated matching.

This will require the backend to save the "last updated time" from the python script in order for the front-end to retrieve it and show it according to the following:

## AC

- [ ] We should run the python script to recalculate the estimated matching for each project once per 10 minutes for the first 4 hours (after the round start time), and then just once per hour for the rest of the round until the end time.

- [ ] The backend should return a value (timestamp) on the last time the script was run and give it to the front end to show throughout the dapp @CarlosQ96 

During an active QF (after the "start time" and before the "end time"), for all projects in that QF round, we should show the last updated time, as for

- [ ] @kkatusic On the project card (appearing for QF projects both on https://giveth.io/projects/all and https://giveth.io/qf/all), the "last updated" time should appear within the "estimated matching tooltip", adding the clock icon and the copy "Estimated matching updated < x > min ago", as follows:

![Image](https://github.com/user-attachments/assets/9017efd3-2a06-4eae-b24b-6ba1a6845dbf)
![Image](https://github.com/user-attachments/assets/f5ea66a4-8da2-4a04-9a4d-355c01501311)

- [ ] @kkatusic  Similarly, on the donate page, the tooltip copy should be adjusted the same, showing the last updated time grabbed from the backend:

![Image](https://github.com/user-attachments/assets/f730f155-f2e3-4c20-a75b-d1c5b1089222)

- [ ] @kkatusic On the project single page (for a project in the active QF round), the estimated matching box should have a new addition at the bottom, showing the last updated time - @mosaeedi can you put the figma link here please?

![Image](https://github.com/user-attachments/assets/ddc23ea6-76f2-472f-a064-931e361c0c5d)

- [ ] @kkatusic On the project single page donations tab, the estimated matching box should have a new addition at the bottom, showing the last updated time - @mosaeedi can you put the figma link here please?

<img width="584" alt="Image" src="https://github.com/user-attachments/assets/0e82ae87-f978-437b-afb5-2801c2de6891" />

**Comment by kkatusic** [comment]
@laurenluz, cc @divine-comedian, this is ready to test

**Comment by maryjaf** [comment]
- [ ] 1- in donation page, this message is shown: "estimated matching updated 0 min ago"

![Image](https://github.com/user-attachments/assets/8770fdd2-bc70-42de-87d0-0e8833e2e236)


and one question, what is the period of updating estimated matching?

@kkatusic

**Comment by kkatusic** [comment]
@maryjaf period is from last time cron job run estimated matching, if you refresh after 4 minutes it should be 4 minutes ago

**Comment by kkatusic** [comment]
@maryjaf when you find time you can test it again on donation page, thx

**Comment by maryjaf** [comment]
> [@maryjaf](https://github.com/maryjaf) when you find time you can test it again on donation page, thx

Thanks @kkatusic Resolved
